cmd/consumer/main.go

```diff
@@ -143,7 +143,7 @@ func appendToFile(filename, text string) error {
 func (rp *RecordProcessor) Shutdown(checkpointer kcl.Checkpointer, reason string) error {
 	if reason == "TERMINATE" {
 		fmt.Fprintf(os.Stderr, "Was told to terminate, will attempt to checkpoint.\n")
-		rp.firehoseWriter.FlushAll()
+		rp.firehoseWriter.Shutdown()
 		rp.checkpoint(checkpointer, "", 0)
```

Comment:
"Calling the new KCL checkpoint operation checkpoint() without any arguments is semantically equivalent to checkpointing the sequence number of the last Record call in the batch, along with the implicit subsequence number 0 (zero)."

http://docs.aws.amazon.com/streams/latest/dev/kinesis-kpl-consumer-deaggregation.html